modules/postgres-cloud-native/main.tf

```diff
@@ -54,14 +71,14 @@ resource "kubernetes_secret" "connection-secret" {
 
   data = {
     "dbname"     = "application-database"
-    "host"       = "${var.argo_deployment_name}-cluster-rw.${var.namespace}"
-    "jdbc-uri"   = "jdbc:postgresql://${var.argo_deployment_name}-cluster-rw.${var.namespace}:5432/application-database?password=${random_password.pg-password.result}&user=application-database"
+    "host"       = "${var.argo_deployment_name}-pooler-rw.${var.namespace}"
+    "jdbc-uri"   = "jdbc:postgresql://${var.argo_deployment_name}-pooler-rw.${var.namespace}:5432/application-database?password=${random_password.pg-password.result}&user=application-database"
     "password"   = random_password.pg-password.result
-    "pgpass"     = "${var.argo_deployment_name}-cluster-rw:5432:application-database:application-database:${random_password.pg-password.result}"
+    "pgpass"     = "${var.argo_deployment_name}-pooler-rw:5432:application-database:application-database:${random_password.pg-password.result}"
     "port"       = "5432"
-    "uri"        = "postgresql://application-database:${random_password.pg-password.result}@${var.argo_deployment_name}-cluster-rw.${var.namespace}:5432/application-database"
+    "uri"        = "postgresql://application-database:${random_password.pg-password.result}@${var.argo_deployment_name}-pooler-rw.${var.namespace}:5432/application-database"
     "user"       = "application-database"
     "username"   = "application-database"
-    "connection" = "postgresql://application-database:${random_password.pg-password.result}@${var.argo_deployment_name}-cluster-rw.${var.namespace}:5432/application-database"
+    "connection" = "postgresql://application-database:${random_password.pg-password.result}@${var.argo_deployment_name}-pooler-rw.${var.namespace}:5432/application-database"

```

Comment:
A future enhancement would be to allow the person using this module to create a connection pooler or not. Likely they would have a sensible application that doesn't take every connection to the DB possible.